client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-styles.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button } from '@automattic/components';
+import { NavigatorButtonAsItem, NavigatorHeader, NavigatorItemGroup } from '@automattic/onboarding';
+import {
+	__experimentalVStack as VStack,
+	__experimentalUseNavigator as useNavigator,
+} from '@wordpress/components';
+import { color, typography } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { useRef, useEffect, useState } from 'react';
+import { NAVIGATOR_PATHS } from './constants';
+import { PATTERN_ASSEMBLER_EVENTS } from './events';
+import NavigatorTitle from './navigator-title';
+
+interface Props {
+	onMainItemSelect: ( name: string ) => void;
+	onContinueClick: ( callback?: () => void ) => void;
+	recordTracksEvent: ( name: string, eventProperties?: any ) => void;
+	selectedMainItem: string | null;
+	hasColor: boolean;
+	hasFont: boolean;
+}
+
+const ScreenStyles = ( {
+	onMainItemSelect,
+	onContinueClick,
+	recordTracksEvent,
+	selectedMainItem,
+	hasColor,
+	hasFont,
+}: Props ) => {
+	const translate = useTranslate();
+	const [ disabled, setDisabled ] = useState( true );
+	const wrapperRef = useRef< HTMLDivElement | null >( null );
+	const navigator = useNavigator();
+	const headerDescription = translate(
+		'Create your homepage by first adding patterns and then choosing a color palette and font style.'
+	);
+	const getIsActiveButton = ( name: string ) => {
+		return name === selectedMainItem;
+	};
+
+	const togglePanel = () => {
+		if ( selectedMainItem ) {
+			// Toggle panel before continue
+			onMainItemSelect( selectedMainItem );
+		}
+	};
+
+	const handleBackClick = () => {
+		navigator.goBack();
```

Comment:
If we use `goBack` here, people might need to click it multiple times to go back to the previous step.

For example,
1. Click `Colors` to close the Colors panel
2. Click `Colors` again to open the Colors panel
3. Click `Fonts` to open the Fonts panel
4. Click `<` to go back, and you have to click it multiple times

Reply:
Thanks, I reproduced that bug once but I didn't know why

Reply:
How/Where should I handle the back to avoid that?

Reply:
How about squashing and merging https://github.com/Automattic/wp-calypso/pull/80917 into this one? It will resolve this issue immediately

Reply:
Okay, let's merge it with #80917. Note that we still have to emit a Tracks event with custom props per screen `onBack`. Check my recent commits.

Reply:
I've rebased https://github.com/Automattic/wp-calypso/pull/80917 onto this PR again. Feel free to squash and merge the changes into this one when you're ready 🙂